scripts/merge_source_cat.py

```diff
@@ -0,0 +1,415 @@
+
+    columns_to_keep = list(DummyDC2SourceCatalog(dm_schema_version).required_native_quantities)
+
+    collected_cats = pd.DataFrame()
+    for dr in data_refs:
+        if not dr.datasetExists():
+            if verbose:
+                print("Skipping non-existent dataset: ", dr.dataId)
+            continue
+
+        if verbose:
+            print("Processing ", dr.dataId)
+        src_cat = load_detector(dr, object_table=object_table,
+                                columns_to_keep=columns_to_keep,
+                                verbose=verbose, **kwargs)
+        if len(src_cat) == 0:
+            if verbose:
+                print("  No good entries for ", dr.dataId)
+            continue
+        collected_cats = collected_cats.append(src_cat)
+
+    if overwrite:
+        if os.path.exists(filename):
+            os.remove(filename)
+
+    if len(collected_cats) == 0:
+        if verbose:
+            print("No sources collected from ", data_refs.dataId)
+            return
+
+    collected_cats.to_parquet(filename)
+
+
+def load_detector(data_ref, object_table=None, matching_radius=1,
+                  columns_to_keep=None, debug=False, **kwargs):
+    """Load detector source catalog and associate sources with Object Table.
+
+    Parameters
+    --
+    data_ref:
+        Butler data_ref
+    object_table:  AFW Table, Pandas DataFrame, or AstroPy Table
+        Table of at least ('id', 'ra', 'dec') for spatial matching to assign
+        the associated Object Id.
+    matching_radius:  float [arcsec]
+    columns_to_keep:  iterable
+        If None, then keep all columns
+
+    Returns
+    --
+    Pandas DataFrame of all sources for a visit in one catalog
+    with photometric calibration and associated Object
+    """
+    cat = data_ref.get(datasetType='src')
+
+    flux_field_names_per_schema_version = {
+        1: {'psf_flux': 'base_PsfFlux_flux', 'psf_flux_err': 'base_PsfFlux_fluxSigma'},
+        2: {'psf_flux': 'base_PsfFlux_flux', 'psf_flux_err': 'base_PsfFlux_fluxErr'},
+        3: {'psf_flux': 'base_PsfFlux_instFlux', 'psf_flux_err': 'base_PsfFlux_instFluxErr'}
+    }
+
+    if debug:
+        print("AFW photometry catalog schema version: {}".format(cat.schema.VERSION))
+    flux_names = flux_field_names_per_schema_version[cat.schema.VERSION]
+
+    # Get a Pandas DataFrame out that we can more easily manipulate:
+    cat = cat.asAstropy().to_pandas()
+
+    # Add visit, filter information as columns.
+    # There's no separate metadata field so we redundantly include here
+    # In the Parquet storage these columns will be categoricals and not take up much disk space.
+    # Pandas automatically broadcasts a scalar to be the same shape as the DataFrame.
+    cat['visit'] = data_ref.dataId['visit']
+    cat['detector'] = data_ref.dataId['detector']
+    cat['filter'] = data_ref.dataId['filter']
+
+    # Calibrate magnitudes and fluxes
+    calib = data_ref.get('calexp_calib')
+    calib.setThrowOnNegativeFlux(False)
```

Comment:
Will there be (an option for) some kind of detection cut?

Reply:
A SNR threshold on the Source can be done after the fact when manipulating the Source Catalog.

This here is just to prevent the wrapped C++ code from throwing an exception when taking the log of a negative number.  I'm actually planning on moving the calibration stuff out of the extraction and just recording the `fluxmag0` in future work..

Reply:
I see. Thank you!